src/Integration.UnitTests/Connection/ConnectionWorkflowTests.cs

```diff
@@ -544,7 +374,8 @@ public async Task ConnectionWorkflow_ConnectionStep_10KProjectsAndKeyIsNotPartOf
 
             AssertCredentialsStored(connectionInfo);
 
-            logger.AssertOutputStrings(4);
```

Comment:
Checking for a specific number of output messages is quite fragile and it's not something we usually do anymore.
I've replaced it with a more specific check in this case that a message about the expected project has been logged.